eth/beacon/types/validator_records.py

```diff
@@ -95,9 +100,12 @@ def get_pending_validator(cls,
             withdrawal_credentials=withdrawal_credentials,
             randao_commitment=randao_commitment,
             randao_layers=0,
-            status=ValidatorStatusCode.PENDING_ACTIVATION,
-            latest_status_change_slot=latest_status_change_slot,
+            activation_slot=FAR_FUTURE_SLOT,
+            exit_slot=FAR_FUTURE_SLOT,
+            withdrawal_slot=FAR_FUTURE_SLOT,
```

Comment:
I assume that we are using `FAR_FUTURE_SLOT` to represent the concept of *undefined* or `None` since it works out to something like 5 billion centuries in the future?

Reply:
Yes, it is essentially an "unset" flag on the var. It works nicely because we can use it in status checks.'
example: is_active -- `v.activation_slot <= state.slot < v.exit_slot`
* in the case that neither `activation_slot` nor `exit_slot` have been updated, the validator is not active.
* in the case that `activation_slot` is set but is in the future, the validator is non active
* in the case that `activation_slot` is in the past, but `exit_slot` is not yet set, then the validator is active
* in the case that `activation_slot` is in the past` and `exit_slot` has been set but is still in the future, the validator is active
* in the case that `activation_slot` and `exit_slot` are in the past, then the validator is not active and is instead exited.

Reply:
Is there any way that a validator could set any of these three fields to this value themselves?  If so, it might be useful to explicitely define `2**64 - 1` as a disallowed value so that we have strong assurances that when any of these values is equal to `2**64 -1` we **know** that it means unset.

Reply:
Yeah, that seems reasonable.

These values can only change via state transitions that a validator can trigger but not control the exact value of (e.g. requesting and `exit` which would set their `exit_slot` as `state.slot + ENTRY_EXIT_DELAY`, or getting penalized which would set their`penalization_slot` to `state.slot`). These state transitions all change the value from `FAR_FUTURE_SLOT` to a value that is a function of the current `state.slot`